PaulasWeatherApp/PaulasWeatherApp/ViewController/CurrentWeatherViewController.swift

```diff
@@ -8,11 +8,108 @@
+    private let sunsetLabel = UILabel()
+    
+    private var sunsetString = NSMutableAttributedString()
+    private var sunriseString = NSMutableAttributedString()
+    
+    let activityIndicator = UIActivityIndicatorView(style: .large)
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
+        setActivityIndicator()
+        
+        sunsetString = createString(for: "sunset")
+        sunriseString = createString(for: "sunrise")
+        
+        setTempLabel()
+        setSunriseLabel()
+        setSunsetLabel()
         view.backgroundColor = .systemRed
     }
     
+    private func setActivityIndicator() {
+        activityIndicator.center = view.center
+        activityIndicator.hidesWhenStopped = true
+        view.addSubview(activityIndicator)
+        activityIndicator.startAnimating()
+    }
+    
+    private func setTempLabel() {
+        
+        tempLabel.translatesAutoresizingMaskIntoConstraints = false
+        tempLabel.textAlignment = .center
+        tempLabel.font = .systemFont(ofSize: 36)
+        tempLabel.textColor = .black
+        tempLabel.text = "... °C"
+        tempLabel.backgroundColor = .clear
+        tempLabel.sizeToFit()
+        
+        view.addSubview(tempLabel)
+        
+        NSLayoutConstraint.activate([
+            tempLabel.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            tempLabel.centerYAnchor.constraint(equalTo: view.centerYAnchor, constant: -70),
+        ])
+    }
+    
+    private func setSunsetLabel() {
+        sunsetLabel.translatesAutoresizingMaskIntoConstraints = false
+        sunsetLabel.textAlignment = .center
+        sunsetLabel.font = .systemFont(ofSize: 12)
+        sunsetLabel.textColor = .black
+        sunsetLabel.backgroundColor = .clear
+        sunsetLabel.attributedText = sunsetString
+        sunsetLabel.sizeToFit()
+        
+        view.addSubview(sunsetLabel)
+        
+        NSLayoutConstraint.activate([
+            sunsetLabel.leadingAnchor.constraint(equalTo: sunriseLabel.trailingAnchor,constant: 10),
+            sunsetLabel.trailingAnchor.constraint(equalTo: tempLabel.trailingAnchor),
+            sunsetLabel.topAnchor.constraint(equalTo: tempLabel.bottomAnchor)
+        ])
+    }
+    
+    private func setSunriseLabel() {
+        sunriseLabel.translatesAutoresizingMaskIntoConstraints = false
+        sunriseLabel.textAlignment = .center
+        sunriseLabel.font = .systemFont(ofSize: 12)
+        sunriseLabel.textColor = .black
+        sunriseLabel.backgroundColor = .clear
+        sunriseLabel.attributedText = sunriseString
+        sunriseLabel.sizeToFit()
+        view.addSubview(sunriseLabel)
+        
+        NSLayoutConstraint.activate([
+            sunriseLabel.leadingAnchor.constraint(equalTo: tempLabel.leadingAnchor),
+            sunriseLabel.topAnchor.constraint(equalTo: tempLabel.bottomAnchor)
+        ])
+    }
+    
```

Comment:
Just a suggestion. It's nice to group methods so when you have a lot, to know where to look. You can use:
// MARK: - Public methods
for example, and you'll see in the navigation of a class that they are very nice divided. The separation logic is really up to you, but it's a nice way to make the code easy to understand and maintain.